db/migrate/20170629151616_add_unique_index_to_repos.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToRepos < ActiveRecord::Migration[5.1]
+  def change
+    add_index :repos, [:name, :user_name], :unique => true
```

Comment:
note that a `CONSTRAINT` implies an index, and the other way around, as far as I know